### PR TITLE
Refs [TASK][36458] list drink datasource

### PR DESCRIPTION
**Pull request by VietKFC**
## Related Tickets
- [#Task 36458: List drink datasource](https://edu-redmine.sun-asterisk.vn/issues/36458)
## WHAT
-
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://github.com/framgia/coding-standards/tree/master/eng/android |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Redmine | Sun* Redmine working process  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md |<li>- [ ] yes</li>|<li>- [ ] yes</li>

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/example/cocktailmaster/data/source/remote/api/APIQuery.kt [hunk 1]
```diff
@@ -18,22 +18,54 @@ object APIQuery {
...
         .appendPath(APIConstant.API_CONTENT)
         .appendPath(APIConstant.API_SEARCH)
-        .appendQueryParameter(ParameterConstant.SEARCH_PARAM , query)
+        .appendQueryParameter(ParameterConstant.SEARCH_PARAM, query)
+        .toString()
+
+    fun filterByCategory(category: String) = Uri.Builder()
+        .scheme(APIConstant.SCHEME_API)
+        .authority(APIConstant.AUTHORITY)
+        .appendPath(APIConstant.API_CONTENT)
```
Comment (tuannm-1276) [inline app/src/main/java/com/example/cocktailmaster/data/source/remote/api/APIQuery.kt]: những thằng này ở Phương thức nào cũng có, e có viết 1 phương thức trả về 1 String sau gắn vào các thằng khác ko

Comment (VietKFC) [reply app/src/main/java/com/example/cocktailmaster/data/source/remote/api/APIQuery.kt]: e da viet roi a

File: app/src/main/java/com/example/cocktailmaster/data/source/remote/utils/ApiFieldConstant.kt [hunk 2]
```diff
@@ -8,14 +8,29 @@ object DrinkConstant {
...
+        "strIngredient13",
+        "strIngredient14",
+        "strIngredient15"
+    )
+    val MEASUREMENTS = listOf(
+        "strMeasure1", "strMeasure2", "strMeasure3", "strMeasure4",
+        "strMeasure5", "strMeasure6", "strMeasure7", "strMeasure8", "strMeasure9",
+        "strMeasure", "strMeasure11", "strMeasure12", "strMeasure13", "strMeasure14",
+        "strMeasure15"
+    )
```
Comment (thonc-2135) [inline app/src/main/java/com/example/cocktailmaster/data/source/remote/utils/ApiFieldConstant.kt]: format 1 thứ 1 kiểu v

File: app/src/main/java/com/example/cocktailmaster/ui/listdrink/ListDrinkPresenter.kt [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            }
+
+            override fun onFailed() {
+                view.hideLoading()
+                view.showError()
+            }
+        })
+    }
+
+    override fun excute() {
```
Comment (thonc-2135) [inline app/src/main/java/com/example/cocktailmaster/ui/listdrink/ListDrinkPresenter.kt]: xem lại chính tả

**Pull request merged**